### ISSUE TITLE
Lazier operation?

### ISSUE THREAD

**Issue by mkitti** [tshort/Eyeball.jl#38]
I have some large JSON files with 300K lines (200 MB). Would it be possible to have eyeball be lazy and only load the one depth level at a time rather than everything eagerly.

**Comment by tshort** [comment]
Yeah. I'm running into that now, too. I looked a bit, and I don't see an obvious easy fix to get that.

**Comment by mkitti** [comment]
How about a `LazyEyeball.jl`?

**Comment by tshort** [comment]
I'm not opposed!

**Comment by tshort** [comment]
Lazy really should be the default here. Not sure how hard that is to accomplish with this code base.
